packages/cli/lib/lib/entry.js

```diff
@@ -14,17 +12,17 @@ if (process.env.NODE_ENV === 'development') {
 	// only add a debug sw if webpack service worker is not requested.
 	if (process.env.ADD_SW === undefined && 'serviceWorker' in navigator) {
 		// eslint-disable-next-line no-undef
-		navigator.serviceWorker.register(__webpack_public_path__ + 'sw-debug.js');
+		navigator.serviceWorker.register(process.env.SW_PATH + 'sw-debug.js');
 	} else if (process.env.ADD_SW && 'serviceWorker' in navigator) {
 		// eslint-disable-next-line no-undef
 		navigator.serviceWorker.register(
-			__webpack_public_path__ + (process.env.ES_BUILD ? 'sw-esm.js' : 'sw.js')
+			process.env.SW_PATH + (process.env.ES_BUILD ? 'sw-esm.js' : 'sw.js')
```

Comment:
Why cant we 
```js
// Service workers are not allowed to be loaded from external domain
navigator.serviceWorker.register(__webpack_public_path__.startsWith('https://')?'':__webpack_public_path__) + (process.env.ES_BUILD ? 'sw-esm.js' :'sw.js')
```

Reply:
I guess, I just thought that we shouldn't assume that it's going to be served from root and give some flexibility for setting the sw path.

Reply:
@prateekbh , removed the flag, and added a check for both http + https protocol.